tests/cpp/test_main.cc

```diff
@@ -3,9 +3,13 @@
 #include <xgboost/base.h>
 #include <xgboost/logging.h>
 #include <string>
+#include <memory>
 #include <vector>
 
+#include "helpers.h"
+
 int main(int argc, char ** argv) {
+  auto rmm_alloc = xgboost::SetUpRMMResourceForCppTests();
```

Comment:
Why do we need manual setup?  Is it required for running normal tasks instead of tests?

Reply:
No, this is to ensure that testxgboost is run with RMM pool allocator instead of the usual CUDA allocator.

Reply:
Will ping you offline next week. This is confusing.

Reply:
The RMM allocator can be configured in multiple configurations. By default, RMM uses `rmm::mr::cuda_memory_resource`, which is just a wrapper around `cudaMalloc()`. The pool allocator (`rmm::mr::pool_memory_resource`) has potential to speed up allocations, and this allocator must be explicitly configured. It can be configured in two ways:
* Call `rmm::mr::set_default_resource()` in C++.
* Call `rmm.mr.set_default_resource()` in Python.